languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -115936,6 +115936,1297 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+                            <exception>affaire</exception>
+                            <exception>assurance</exception>
+                            <exception>barre</exception>
+                            <exception>barres</exception>
+                            <exception>besoin</exception>
+                            <exception>bobo</exception>
+                            <exception>carte</exception>
+                            <exception>chaud</exception>
+                            <exception>cœur</exception>
+                            <exception>connaissance</exception>
+                            <exception>conscience</exception>
+                            <exception>cours</exception>
+                            <exception>coutume</exception>
+                            <exception>droit</exception>
+                            <exception>envie</exception>
+                            <exception>faim</exception>
+                            <exception>foi</exception>
+                            <exception>frais</exception>
+                            <exception>froid</exception>
+                            <exception>grâce</exception>
+                            <exception>intuition</exception>
+                            <exception>hâte</exception>
+                            <exception>honte</exception>
+                            <exception>horreur</exception>
+                            <exception>idée</exception>
+                            <exception>lieu</exception>
+                            <exception>maille</exception>
+                            <exception>mal</exception>
+                            <exception>martel</exception>
+                            <exception>nom</exception>
+                            <exception>part</exception>
+                            <exception>peur</exception>
+                            <exception>pied</exception>
+                            <exception>pignon</exception>
+                            <exception>pitié</exception>
+                            <exception>prise</exception>
+                            <exception>raison</exception>
+                            <exception>recours</exception>
+                            <exception>soif</exception>
+                            <exception>souci</exception>
+                            <exception>trait</exception>
+                            <exception>vent</exception>
+                            <exception>voix</exception>
+                        </token>
+                    </marker>
+                </pattern>
+                <message>Après le verbe 'avoir', ce nom n'est généralement pas utilisé.</message>
+                <suggestion>un \2</suggestion>
+                <suggestion>une \2</suggestion>
+                <suggestion>des \2</suggestion>
+                <example correction="un voiture|une voiture|des voiture">J'ai <marker>voiture</marker>.</example>
```

Comment:
**question(blocking):** What's intended here? On the sentence "j'ai fée des pâtes", for example, the rule would produce suggestions leading to "j'ai un|une|des fée des pâtes". How is that useful for the users, especially compared to the very high rate of false positives I can already see (j'ai piscine, poney, clarinette...).

Reply:
You're right that the suggestion is not so relevant in that context, however, it usefully drags the attention of the user to the problem that "fée" is a noun. Also, "j'ai une fée des pâtes" is grammatically correct, provided that such a fairy exist (or that you are married to a skilled pasta cooker ^^).
Anyway, generating suggestions for those errors will always be tricky, but detecting the error is already very useful.

Your point about "j'ai piscine" and such is a very strong point. We might have to give up the rule about "avoir", just like we did with "parler", because too many words could go there. We could, still detect some nouns that should not go there, like "fée", for people with dyslexia.

Tell me if that strategy would seem ok to you.

Reply:
Yes, that could work.
Please ensure to do the other changes requested 💪